From 301ea1eba38f66dce3138d63544e9d7e00380d6b Mon Sep 17 00:00:00 2001
From: Yannick Lanz <yannick.lanz@wifx.net>
Date: Mon, 31 Mar 2025 11:16:29 +0200
Subject: [PATCH 05/11] mc610: remove useless check after SIM unlock (modem
 starts now in low power mode and doesn't timeout during initial phase

---
 .../mm-broadband-modem-fibocom-mc610.c        | 48 -------------------
 1 file changed, 48 deletions(-)

diff --git a/src/plugins/fibocom_mc610/mm-broadband-modem-fibocom-mc610.c b/src/plugins/fibocom_mc610/mm-broadband-modem-fibocom-mc610.c
index 9a9d1635..cf047c3f 100644
--- a/src/plugins/fibocom_mc610/mm-broadband-modem-fibocom-mc610.c
+++ b/src/plugins/fibocom_mc610/mm-broadband-modem-fibocom-mc610.c
@@ -238,49 +238,6 @@ modem_power_off (MMIfaceModem *self,
                               user_data);
 }
 
-/*****************************************************************************/
-/* After SIM unlock (Modem interface) */
-
- static gboolean
- modem_after_sim_unlock_finish (MMIfaceModem *self,
-                                GAsyncResult *res,
-                                GError **error)
- {
-     return g_task_propagate_boolean (G_TASK (res), error);
- }
-
- static void
- cpin_try_ready (MMBaseModem *self,
-                 GAsyncResult *res,
-                 GTask *task)
- {
-     /* Well, we're done */
-     g_task_return_boolean (task, TRUE);
-     g_object_unref (task);
- }
-
- static void
- modem_after_sim_unlock (MMIfaceModem *self,
-                         GAsyncReadyCallback callback,
-                         gpointer user_data)
- {
-     GTask *task;
-     task = g_task_new (self, NULL, callback, user_data);
-
-     /* Fibocom MC610 can be very long after SIM is enabled as it proceed
-      * directly with network attachement if SIM allow it (most cases).
-      * +CPIN? is a know command which will wait on network attachement.
-      */
-     /* TODO: This is probably not the case anymore as we now are in flightmode
-       * during initialization. Test if this is still needed.
-       */
-     mm_base_modem_at_command (MM_BASE_MODEM (self),
-                               "+CPIN?",
-                               30,
-                               FALSE,
-                               (GAsyncReadyCallback)cpin_try_ready,
-                               task);
- }
 
  /*****************************************************************************/
 /* Load supported modes (Modem interface) */
@@ -732,11 +689,6 @@ iface_modem_init (MMIfaceModemInterface *iface)
     iface->modem_power_off = modem_power_off;
     iface->modem_power_off_finish = modem_common_power_finish;
 
-#if 0
-    iface->modem_after_sim_unlock = modem_after_sim_unlock;
-    iface->modem_after_sim_unlock_finish = modem_after_sim_unlock_finish;
-#endif
-
     iface->load_supported_modes = load_supported_modes;
     iface->load_supported_modes_finish = load_supported_modes_finish;
     iface->load_current_modes = load_current_modes;
-- 
2.45.2

